content/concepts/the-rippled-server/ledger-history/online-deletion.md

```diff
@@ -0,0 +1,107 @@
+# Online Deletion
+[[Source]<br/>](https://github.com/ripple/rippled/blob/master/src/ripple/app/misc/SHAMapStoreImp.cpp "Source")
+
+The online deletion feature lets the `rippled` server delete the server's local copy of old ledger versions to keep disk usage from rapidly growing over time. The default config file sets online deletion to run automatically, but online deletion can also be configured to run only when prompted. [New in: rippled 0.27.0][]
+
+The server always keeps the complete _current_ state of the ledger, with all the balances and settings it contains. The deleted data includes older transactions and versions of the ledger state that are older than the stored history.
+
+The default config file sets the `rippled` server to keep the most recent 2000 ledger versions and automatically delete older data.
+
+**Tip:** Even with online deletion, the amount of disk space required to store the same time span's worth of ledger data increases over time, because the size of individual ledger versions tends to grow over time. This growth is very slow in comparison to the accumulation of data that occurs without deleting old ledgers. For more information on disk space needs, see [Capacity Planning](capacity-planning.html).
+
+
+## Background
+
+The `rippled` server stores [ledger history](ledger-history.html) in its _ledger store_. This data accumulates over time.
+
+Inside the ledger store, ledger data is "de-duplicated": data that doesn't change from version to version is only stored once. The records themselves in the ledger store do not indicate which ledger version(s) contain them; part of the work of online deletion is identifying which records are only used by outdated ledger versions. This process is time consuming and affects the disk I/O and application cache, so it is not feasible to delete old data on every ledger close.
```

Comment:
Nit - stray ":"? 

>ledger data is "de-duplicated": data

to

>ledger data is "de-duplicated" data

Reply:
Is this a valid suggested edit? 🤔 

>is "de-duplicated" data that doesn't change from version to version is only stored once.

to

>is "de-duplicated" data that doesn't change from version to version and is only stored once.

Reply:
@jhaaaa Your suggested edit is valid, but I actually prefer the way it was written. I like short sentences, and the colon splits one long sentence into two short ones, but this is just my opinion.

Reply:
The colon was not stray. It indicates that the remainder is an independent clause that clarifies the preceding clause. I see where you mis-read "de-duplicated" as being an adjective that modifies the following data, but it's actually a predicate adjective that modifies the previous instance of "data".

I don't like connecting the clauses with "that" because it implies that there is another meaning or use of "de-duplicated".  To make it easier for others to avoid the same mistake, I've rephrased it into two separate sentences:

> Inside the ledger store, ledger data is "de-duplicated". In other words, data that doesn't change from version to version is only stored once.